be/src/pipeline/dependency.cpp

```diff
@@ -188,10 +188,12 @@ void LocalExchangeSharedState::sub_running_sink_operators() {
     }
 }
 
-void LocalExchangeSharedState::sub_running_source_operators() {
+void LocalExchangeSharedState::sub_running_source_operators(
+        LocalExchangeSourceLocalState& local_state) {
     std::unique_lock<std::mutex> lc(le_lock);
     if (exchanger->_running_source_operators.fetch_sub(1) == 1) {
         _set_always_ready();
```

Comment:
This API will also set sink depdendency to always ready.
If it is always ready, then the sink pipeline may add many many blocks to the queue,  maybe it will OOM?

Reply:
Data queue will be set EOS if `_running_source_operators == 0` so sink operators will never put a new block into data queue.